### PR TITLE
Add 'dns_dyn' DNS challenge validation script for Dyn Managed DNS API, Issue #594

### PR DESCRIPTION
**Pull request by abelbeck**
Add 'dns_dyn' DNS challenge validation script for Dyn Managed DNS API

Original Author: Gerd Naschenweng
https://github.com/magicdude4eva

### CODE REVIEW

**Comment by Neilpang** [comment]
It seems that  shellcheck has updated it's default rules,  I'm fixing.

File: dnsapi/dns_dyn.sh [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+  export _H2="Content-Type: application/json"
+
+  response="$(_get "$dyn_url" "" "")"
+  sessionstatus="$(printf "%s\n" "$response" | _egrep_o '"status" *: *"[^"]*' | head -n 1 | sed 's#^"status" *: *"##')"
+
+  _debug response "$response"
+  _debug sessionstatus "$sessionstatus"
+
+  if [ "$sessionstatus" = "success" ]; then
+    _dyn_record_id="$(printf "%s\n" "$response" | _egrep_o "\"data\" *: *\[\"/REST/TXTRecord/$_dyn_zone/$fulldomain/[^\"]*" | head -n 1 | sed "s#^\"data\" *: *\[\"/REST/TXTRecord/$_dyn_zone/$fulldomain/##")"
```
Comment (Neilpang) [inline dnsapi/dns_dyn.sh]: please use `_head_n()` function instead.

**Comment by Neilpang** [comment]
Please rebase code.
And also update the main readme file:  https://github.com/Neilpang/acme.sh/blob/master/README.md
List the in the api list.

Thanks.

**Comment by abelbeck** [comment]
@Neilpang The "Files changed" here looks good, but if I can do anything make merging cleaner for you, let me know, I not a git guru.

**Comment by Neilpang** [comment]
Thank you.

**Pull request merged**